### PR TITLE
Add Interceptor support for PUSH_REGISTER and PUSH_AUTHENTICATE Action types 

### PR DESCRIPTION
**Pull request by vahancouver**
[SDKS-2545](https://bugster.forgerock.org/jira/browse/SDKS-2545)

Add Interceptor support for PUSH_REGISTER and PUSH_AUTHENTICATE Action types 

Add unit tests

### CODE REVIEW

File: FRAuthenticator/FRAuthenticatorTests/UnitTests/RequestInterceptor/RequestInterceptorTests.swift [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        catch {
+            XCTFail("Push authentication failed to prepare auth request")
+        }
+    }
+}
+
+class PushRequestInterceptor: RequestInterceptor {
+    func intercept(request: Request, action: Action) -> Request {
+        if action.type == "PUSH_REGISTER" {
+            RequestInterceptorTests.intercepted.append("PUSH_REGISTER")
```
Comment (jeyanthanperiyasamy) [inline FRAuthenticator/FRAuthenticatorTests/UnitTests/RequestInterceptor/RequestInterceptorTests.swift:119]: as an extra step , i think we can add some custom header add those custom header part of AM_Push_Registration_Successful mocks to make sure the request has custom header .

**Review by jeyanthanperiyasamy — CHANGES_REQUESTED** [review]
just make sure this changes will work all type of authentication like number challenge, biometric authentication, accept from the  push notification with or without opening the app

File: FRAuthenticator/FRAuthenticatorTests/UnitTests/RequestInterceptor/RequestInterceptorTests.swift (hunk 1 above)
Comment (vahancouver) [reply FRAuthenticator/FRAuthenticatorTests/UnitTests/RequestInterceptor/RequestInterceptorTests.swift:145]: Updated the tests. added custom header to the request.

**Comment by vahancouver** [comment]
> just make sure this changes will work all type of authentication like number challenge, biometric authentication, accept from the push notification with or without opening the app

Updated the test to use more generic handleNotification method that is called for accepting/denying all types of push notifications

**Review by jeyanthanperiyasamy — APPROVED** [review]

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

**Review by spetrov — APPROVED** [review]
Looks good! 👍🏻

**Pull request merged**